tests/unit/test_logging.py

```diff
@@ -75,3 +91,103 @@ def test_step_number(step: int) -> None:
+        data, models = {"A": mk_dataset([[0.0]], [[0.0]])}, {"A": _PseudoTrainableQuadratic()}
+        steps = 5
+        rule = FixedAcquisitionRule([[0.0]])
+        BayesianOptimizer(lambda x: {"A": Dataset(x, x ** 2)}, Box([-1], [1])).optimize(
+            steps, data, models, rule
+        )
+
+    ordered_scalar_names = [
+        "A.observation.best_overall",
+        "A.observation.best_new",
+        "Wallclock.total",
+        "Wallclock.query_point_generation",
+        "Wallclock.model_fitting",
+    ]
+    N = len(ordered_scalar_names)
+
+    for step in range(steps):  # iterate over saved scalars
+        counter = 0
+        for scalar_name in ordered_scalar_names:
+            assert mocked_summary_scalar.call_args_list[step * N + counter][0][0] == scalar_name
+            assert mocked_summary_scalar.call_args_list[step * N + counter][-1]["step"] == step
+            assert isinstance(mocked_summary_scalar.call_args_list[step + counter][0][1], float)
+            counter += 1
+
+
+@unittest.mock.patch("trieste.models.gpflow.interface.tf.summary.scalar")
+@pytest.mark.parametrize("fit_initial_model", [True, False])
+def test_wallclock_time_logging(
+    mocked_summary_scalar: unittest.mock.MagicMock,
+    fit_initial_model: bool,
+) -> None:
+
+    model_fit_time = 0.1
+    acq_time = 0.05
+
+    class _PseudoTrainableQuadraticWithWaiting(QuadraticMeanAndRBFKernel, PseudoTrainableProbModel):
+        def optimize(self, dataset: Dataset) -> None:
+            sleep(model_fit_time)
+
+    class _FixedAcquisitionRuleWithWaiting(FixedAcquisitionRule):
+        def acquire(
+            self,
+            search_space: SearchSpace,
+            models: Mapping[str, ProbabilisticModel],
+            datasets: Optional[Mapping[str, Dataset]] = None,
+        ) -> TensorType:
+            """
+            :param search_space: Unused.
+            :param models: Unused.
+            :param datasets: Unused.
+            :return: The fixed value specified on initialisation.
+            """
+            sleep(acq_time)
+            return self._qp
+
+    mocked_summary_writer = unittest.mock.MagicMock()
+    with tensorboard_writer(mocked_summary_writer):
+        data, models = {"A": mk_dataset([[0.0]], [[0.0]])}, {
+            "A": _PseudoTrainableQuadraticWithWaiting()
+        }
+        steps = 2
+        rule = _FixedAcquisitionRuleWithWaiting([[0.0]])
+        BayesianOptimizer(lambda x: {"A": Dataset(x, x ** 2)}, Box([-1], [1])).optimize(
+            steps, data, models, rule, fit_initial_model=fit_initial_model
+        )
+
+    for scalar in mocked_summary_scalar.call_args_list:
+        if scalar[0][0][:9] == "Wallclock":
+            assert scalar[0][1] > 0  # want positive wallclock times
+
+    # check logs of first BO step
+    if fit_initial_model:  # logging should count two model fits
+        npt.assert_allclose(
+            mocked_summary_scalar.call_args_list[2][0][1], 2 * model_fit_time + acq_time, rtol=0.1
```

Comment:
Isn't this going to be very fragile? If anyone changes the scalars we log then it'll break, and the fix won't be obvious.

Since this test is only trying to check the values are correct, why not loop over the call args and check the values programmatically? In fact you should be able to just change the loop on line 166 to say something like this instead: (and maybe increase steps to 3 just for good luck)

```python
other_scalars = 0

for scalar in mocked_summary_scalar.call_args_list:
    name = scalar[0][0]
    value = scalar[0][1]
    step = scalar[-1]["step"]
    if name.startswith("Wallclock"):
        assert value > 0  # want positive wallclock times
    if name == "Wallclock.total":
        npt.assert_allclose(value, model_fit_time + acq_time + (model_fit_time if fit_initial_model and step == 0 else 0, rtol=0.1)
    elif name == "Wallclock.query_point_generation"):
        npt.assert_allclose(value, acq_time, rtol=0.1)
    elif name == "Wallclock.model_fitting"):
        npt.assert_allclose(value, model_fit_time + (model_fit_time if fit_initial_model and step == 0 else 0, rtol=0.1)
    else:
        other_scalars += 1

# check that we processed all the wallclocks we were expecting
assert len(mocked_summary_scalar.call_args_list) == other_scalars + 3 * steps
```